File: schichtln/app/src/main/java/com/example/andenk/schichtln/notification/Message.kt
package com.example.andenk.schichtln.notification

import com.example.andenk.schichtln.helper.gson
import com.github.kittinunf.fuel.Fuel
import java.io.Serializable

/**
 * Created by andenk on 29.05.2018.
 */

val SERVERKEY = "AAAAeaDL9ss:APA91bGJJwBuwpBul2mr5zeNBpKOMoJSvFjwV1EMhtFhXtFCoHTjdFr_Eg_MluQ5epIAFDwEY1Ay1i-zpkmiNfXOUDVlYFk_6rMCR1nBQwIwZ4tXoJQRvbV0za4WdgdtKRVIqQC1UMWp"

data class Message(
        var data: MessageData,
        var to: String
) : Serializable

data class MessageData(
        var body: String,
        var title: String
) : Serializable

fun sendMessage(to: String, title: String="CINEMA", body: String) {
    val request = Fuel.post("https://fcm.googleapis.com/fcm/send")
    request.headers["Content-Type"] = "application/json"
    request.headers["Authorization"] = "key=$SERVERKEY"
    val body = gson.toJson(Message(to = "/topics/$to", data = MessageData(body = body, title = title)))
    request.body(body)
            .response { request, response, result ->
                print("awdsfaewf")
            }
}

File: schichtln/ios/schicht_storyboard/schicht_storyboard/New Group/Notifications/Message.swift
//
//  Message.swift
//  schicht_storyboard
//
//  Created by Andreas Denk on 16.07.18.
//  Copyright © 2018 Andreas Denk. All rights reserved.
//

import Foundation
import Alamofire
public struct Message: Codable {
    var data: MessageData
    var to: String
}

public struct MessageData:Codable{
    var title: String
    var body:String
}

public func sendMessage(to: String, title: String="CINEMA", body: String){
    var request = URLRequest(url: URL(string: "https://fcm.googleapis.com/fcm/send")!)

    request.httpMethod = HTTPMethod.post.rawValue
    request.setValue("application/json; charset=UTF-8", forHTTPHeaderField: "Content-Type")
    request.setValue("key=AAAAeaDL9ss:APA91bGJJwBuwpBul2mr5zeNBpKOMoJSvFjwV1EMhtFhXtFCoHTjdFr_Eg_MluQ5epIAFDwEY1Ay1i-zpkmiNfXOUDVlYFk_6rMCR1nBQwIwZ4tXoJQRvbV0za4WdgdtKRVIqQC1UMWp", forHTTPHeaderField: "Authorization")

    let headers: HTTPHeaders = [
        "Authorization": "key=AAAAeaDL9ss:APA91bGJJwBuwpBul2mr5zeNBpKOMoJSvFjwV1EMhtFhXtFCoHTjdFr_Eg_MluQ5epIAFDwEY1Ay1i-zpkmiNfXOUDVlYFk_6rMCR1nBQwIwZ4tXoJQRvbV0za4WdgdtKRVIqQC1UMWp",
        "Content-Type": "application/json"
    ]
    
    let parameters: Parameters = [
        "data": [
            "title": title,
            "body": body
        ],
        "to": "/topics/\(to)"
    ]
    Alamofire.request( "https://fcm.googleapis.com/fcm/send", method: HTTPMethod.post, parameters: parameters, encoding: JSONEncoding.default, headers: headers).responseString{it in
        print(it)
    }
}
